lib/fog/compute/ecloud/requests/ssh_key_create.rb

```diff
@@ -0,0 +1,76 @@
+module Fog
+  module Compute
+    class Ecloud
+      class Real
+        include Shared
+
+        def ssh_key_create(data)
+          validate_data([:name], data)
+
+          request(
+            :body => generate_ssh_key_create_request(data),
+            :expects => 201,
+            :method => "POST",
+            :headers => {},
+            :uri => data[:uri],
+            :parse => true
+          )
+        end
+
+        private
+
+        def generate_ssh_key_create_request(data)
+          xml = Builder::XmlMarkup.new
+          xml.CreateSshKey(:name => data[:name]) do
+            xml.Default data[:default]
+          end
+        end
+      end
+
+      class Mock
+        include Shared
+
+        def ssh_key_create(data)
+          ssh_key_id          = Fog::Mock.random_numbers(7).to_i
+          ssh_key_fingerprint = ""
+          for x in 1..15
+            ssh_key_fingerprint = ssh_key_fingerprint + Fog::Mock.random_hex(2) + ':'
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.